pallets/committee-management/src/impls.rs

```diff
@@ -52,28 +58,64 @@ fn choose_for_session<T: Clone>(validators: &[T], count: usize, session: usize)
     Some(chosen)
 }
 
+/// choose all items from `reserved` if present and extend it by #`non_reserved_seats` from non_reserved if present.
+fn choose_finality_committee<T: Clone>(
+    reserved: &Option<Vec<T>>,
+    non_reserved: &Option<Vec<T>>,
+    non_reserved_seats: usize,
+    session: usize,
+) -> Vec<T> {
+    let non_reserved_finality_committee = non_reserved
+        .as_ref()
+        .and_then(|nr| choose_for_session(nr, non_reserved_seats, session))
+        .unwrap_or_default();
+
+    if let Some(r) = reserved {
```

Comment:
If you want to get rid of the `if`, you can probably start by `let mut finality_committee = reserved.unwrap_or_default();`
and then `extend`.

Reply:
nice :P done